pkg/internal/apis/certmanager/validation/issuer.go

```diff
@@ -284,10 +284,13 @@ func ValidateACMEChallengeSolverDNS01(p *cmacme.ACMEChallengeSolverDNS01, fldPat
 			el = append(el, field.Forbidden(fldPath.Child("clouddns"), "may not specify more than one provider type"))
 		} else {
 			numProviders++
-			// if either of serviceAccount.name or serviceAccount.key is set, we
-			// validate the entire secret key selector
-			if p.CloudDNS.ServiceAccount.Name != "" || p.CloudDNS.ServiceAccount.Key != "" {
-				el = append(el, ValidateSecretKeySelector(&p.CloudDNS.ServiceAccount, fldPath.Child("clouddns", "serviceAccountSecretRef"))...)
+			// if service account is not nil and either of serviceAccount.name
+			// or serviceAccount.key is set, we validate the entire secret key
+			// selector
+			if p.CloudDNS.ServiceAccount != nil {
+				if p.CloudDNS.ServiceAccount.Name != "" || p.CloudDNS.ServiceAccount.Key != "" {
```

Comment:
Why are we checking whether the strings are empty here? An empty string should be the same as a non-empty string for validation in this context. It should be enough to just check `ServiceAccount != nil` I should think. @munnerz 

The rest is looking good to me though :)

Reply:
Well the checks were there before when service account always required so figured they would still be desired in the scenario where a service account was provided. I suppose when someone goes through the trouble of setting a service account it could be considered enough reason to press on with the test rather then checking also whether the name/key is set first. I'll amend.